### DISCUSSION TITLE
Error while warping Unstructured Meshes with Prism Layer (IDWARP)

### DISCUSSION THREAD

**Discussion by BilalAnwar165** [mdolab/dafoam#691] [Bug]
Hi @friedenhe @sseraj 

I am using IDWarp with pyGeo to warp a volume mesh based on the deformations in surface mesh of a wing. I have created mesh in CGNS format and inserted a prism layer as well. The mesh contains Tri, Tetra, Quad and Penta (due to prism layer) elements. The mesh deforms fine witout the prism layer. But when I try to deform the mesh using prism layer it outputs the error or message that "**element must be a fixed size**". Does IDWarp can handle meshes with prism layer? please advise

[
![error_](https://github.com/user-attachments/assets/1ca0f296-a2b1-470b-a231-f2300b400f73)
](url)

**Comment by friedenhe** [comment]
This seems to be an IDWarp issue. And I do not have a solution to this... What you can do is use DAFoam framework to deform the mesh.

**Reply by BilalAnwar165** [reply]
Thankyou for your reply @friedenhe . Can DAFoam framework deform unstructured (tri/tetra) meshes with Prism layer?

**Comment by friedenhe** [comment]
Yes.
